src/platform/ESP32/ESP32FactoryDataProvider.h

```diff
@@ -32,16 +32,16 @@ class ESP32FactoryDataProvider : public CommissionableDataProvider,
                                  public Credentials::DeviceAttestationCredentialsProvider
 #if CHIP_DEVICE_CONFIG_ENABLE_DEVICE_INSTANCE_INFO_PROVIDER
     ,
-                                 public Internal::GenericDeviceInstanceInfoProvider<Internal::ESP32Config>
+                                 public DeviceInstanceInfoProvider
```

Comment:
Why are we duplicating the code present in `GenericDeviceInstanceInfoProvider`? Methods in `GenericDeviceInstanceInfoProvider` provides the same functionality, and that was the reason to extend that class instead of `DeviceInstanceInfoProvider`.

Reply:
We think it should be better to put them into a file, it is more convenient for customers using.

Reply:
Some customers use the `external platform` feature, who want to customize the getter functions in their own way. So we want to put all these in single file, and customers don't need to modify any code in `GenericDeviceInstanceInfoProvider`.